views/login.jsx

```diff
@@ -0,0 +1,159 @@
+import React from 'react';
+import { Dimensions, ImageBackground, View } from 'react-native';
+import { Container, Header, Content, Card, CardItem, Thumbnail, Text, Button, Icon,
+  Left, Label, Body, Right, Title, Form, Input, Item } from 'native-base';
+import Banner from '../components/Banner'
+import ApiClient from '../ApiClient'
+
+export default class LoginView extends React.Component {
+  static navigationOptions = { header: null };
+
+  constructor(props) {
+    super(props);
+    this.state = { 
+      registering: false,
+      username: null,
+      password: null,
+      firstName: null,
+      lastName: null,
+      goldenTicket: null
+    };
+  }
+
+  toggleRegistering() {
+    this.setState({
+      ...this.state,
+      registering: !this.state.registering,
+      errorMessage: null,
+    });
+  }
+
+  updateFormStateFunc(key) {
+    return function(text) {
+      this.setState({
+        ...this.state,
```

Comment:
@picklechips Just FYI you shouldn't need to do `...this.state` when called `setState`, it'll update just what's provided and persist the rest

Reply:
Both good points ! I'll update that, should make things a little cleaner.